### PR TITLE
completed

### PR DESCRIPTION
**Pull request by Rlemus93**
# Submitting Unit 4 Assessment

### ✋ Reminder

Do NOT merge your assessment!

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅ ] React Challenge
- [✅ ] Interview Practice Questions
- [✅ ] Looking Ahead Questions

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,29 +10,81 @@
...
 
-const fibonacciLength2 = 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const fibonacciSequence = (length) => {
+    if (length <= 2) {
+      return('Length must be greater than 2');
```
Comment (sjproctor) [inline code-challenges.test.js:42]: good check for an edge case!

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,29 +10,81 @@
...
+}) 
+
+//ReferenceError: endTally is not defined
 
 // b) Create the function that makes the test pass.
 
+const endTally = (votes) => {
+  return votes.upVotes - votes.downVotes
+}
+
```
Comment (sjproctor) [inline code-challenges.test.js:85]: Looks great, very clean.

File: interview-questions.md [hunk 3]
```diff
@@ -10,40 +10,40 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 Similarities
 
-1.
-2.
-3.
+1. They both do similiar logic within the code. meaning they both support conditional statements, loops, functions/methods, and others.
+2. both are interpreted languages so code is executed line by line
+3. both are object oriented languages. meaning both support creating and manipulating objects with properties and methods
```
Comment (sjproctor) [inline interview-questions.md:15]: Ruby is definitely object-oriented. JavaScript uses prototypical inheritance so isn't considered a classic oop language.

File: ruby.rb [hunk 4]
```diff
@@ -7,4 +7,47 @@
...
+
+  # Method that takes number as parameter and adds it to current speed which will always start with 0
+  def pedal_faster(number)
+    @current_speed += number
+  end 
+  
+  # Method that takes a number as parameter and adds to the wheels. but you cant exceed more that 3 since we only make tricycles! 
+  def tricycle(number)
+    if @wheels + number > 3
+      p "we only make tricycles!"
```
Comment (sjproctor) [inline ruby.rb:25]: This method looks great but I would make sure the `p` is on the method call down below on line 50 and not on the inner workings of your code.

**Review by sjproctor — APPROVED** [review]
Really great work, Ryan! I really appreciate the depth of your researched answers. You are showing a competent level of skill in this assessment.
